ci/scripts/pr_list_database.py

```diff
@@ -2,154 +2,121 @@
 
 import sys
 import os
-from pathlib import Path
+from wxflow import SQLiteDB
 from argparse import ArgumentParser, ArgumentDefaultsHelpFormatter, REMAINDER, ZERO_OR_MORE
-import sqlite3
 
 
 def full_path(string):
-    """
-    Gets the absolute path of the given file and confirms the directory exists
-
-    Parameters
-    ----------
-    string : str
-        Path to a file
-
-    Returns
-    --------
-    str
-        Absolute path of input path
-
-    Raises
-    -------
-    NotADirectoryError
-        If the target directory for the file does not exist.
-
-    """
-
     if os.path.isfile(string) or os.path.isdir(os.path.dirname(string)):
         return os.path.abspath(string)
     else:
         raise NotADirectoryError(string)
 
 
-def sql_connection(filename: os.path) -> sqlite3.Connection:
+def create(db):
     """
-    Returns an Sqlite3 Cursor object from a given path to a sqlite3 database file
+    Create a new database.
 
     Parameters
     ----------
-    filename : Path
-        Full path to a sqlite3 database file
-
-    Returns
-    -------
-    sqlite3.Connection
-        Sqlite3 Connection object for updating table
-
+    ci_database : SQLiteDB
+        The database to create.
     """
-    try:
-        return sqlite3.connect(filename)
-    except sqlite3.Error:
-        print(sqlite3.Error)
-        sys.exit(-1)
+    db.create_table('pr_list', ['pr INTEGER PRIMARY KEY', 'state TEXT', 'status TEXT', 'reset_id INTEGER', 'cases TEXT'])
 
 
-def sql_table(obj: sqlite3.Cursor) -> None:
+def add_pr(db, pr):
     """
-    Creates the initial sqlite3 table for PR states and status
+    Add a pull request to the database.
 
     Parameters
     ----------
-    obj : sqlite3.Cursor
-         Cursor object for Sqlite3
-
+    ci_database : SQLiteDB
+        The database to add the pull request to.
+    pr : str
+        The pull request to add.
     """
+    rows = db.fetch_data('pr_list')
+    for row in rows:
+        if str(row[0]) == str(args.add_pr[0]):
+            print(f"pr {row[0]} already is in list: nothing added")
+            sys.exit(0)
```

Comment:
I think there is a limitation in this logic.
What happens if a `row` in the list of `rows` is not in the db after a `row` that is in the list?
The logic here will return without checking beyond the current `row`.

Reply:
Yes but this is data base is not really a database but a state machine.  The way it works is that you can only have one and only one PR running at a time and the PR number is unique.  It is this fact that makes the CI work for when some one updates a label of a PR that is already running, it then checks if it should be quit or be restarted.  On the other hand, give that you posed this question.  I can reconsider adding a column that is the state and use that for this logic and impose uniqueness into the database schema.  That would be better.

Reply:
makes sense.  thanks for explaining.